QualityControl/lib/services/QcdbDownload.service.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { LogManager } from '@aliceo2/web-ui';
+import { pipeline } from 'node:stream';
+import { promisify } from 'node:util';
+
+const CONTENT_LENGTH_HEADER = 'Content-Length';
+const CONTENT_TYPE_HEADER = 'Content-Type';
+const CONTENT_DISPOSITION_HEADER = 'Content-Disposition';
+const FILENAME_START_INDEX = 17;
+
+const CONTENT_TYPE_DEFAULT = 'application/root';
+const CONTENT_DISPOSITION_DEFAULT_PARTIAL = 'attachment; filename=';
+const FILENAME_DEFAULT = 'export.root';
+
+/**
+ * @class
+ * Class which sets up the QCDB download service.
+ */
+export class QcdbDownloadService {
+  /**
+   * Constructor
+   * @param {object} config - application's {config.js}.ccdb, needed for QCDB hostname+port
+   */
+  constructor(config = {}) {
+    this._hostname = config.hostname ?? 'localhost';
+    this._port = config.port ?? 8080;
+    this._protocol = config.protocol ?? 'http';
+    this._target = `${this._protocol}://${this._hostname}:${this._port}`;
+
+    this._logger = LogManager.getLogger(`${process.env.npm_config_log_label ?? 'qcg'}/qcdb-download-service`);
+    this._pipelineAsync = promisify(pipeline);
+  }
+
+  /**
+   * Stream the ROOT file contained in the QCDB response into our reponse back to the user.
+   * @param {Response} response - Response from QCDB
+   * @param {Express.Response} res - Outgoing response object we'll write our data into
+   * @returns {void}
+   */
+  async _streamToResponse(response, res) {
+    const contentLength = response.headers.get(CONTENT_LENGTH_HEADER);
+    const contentType = response.headers.get(CONTENT_TYPE_HEADER);
+    const contentDisposition = response.headers.get(CONTENT_DISPOSITION_HEADER);
+    const filename = contentDisposition?.slice(FILENAME_START_INDEX, contentDisposition.length - 1);
+    // We will stream the data from QCDB's answer directly back to the user.
+    res.setHeader(CONTENT_TYPE_HEADER, contentType ?? CONTENT_TYPE_DEFAULT);
+    res.setHeader(CONTENT_DISPOSITION_HEADER, contentDisposition ?? `${CONTENT_DISPOSITION_DEFAULT_PARTIAL}
```

Comment:
```suggestion
    res.setHeader(CONTENT_DISPOSITION_HEADER, contentDisposition ?? `${CONTENT_DISPOSITION_DEFAULT_PARTIAL}"${filename}"`);
```
Turns out quickly responding to an eslint max length issue without my brain turned on might not haven been a good idea.